test: basic boilerplate for metaplex tests

israelidanny opened the pull request:
This copies the basic boilerplate for tests from the JS-SDK.

This is just a quick port, so some cleanup may be needed.

It starts up a local test validator and runs one test successfully.

<img width="992" alt="Screen Shot 2022-01-26 at 03 42 31" src="https://user-images.githubusercontent.com/1970424/151083656-12ad551f-352c-483a-8577-790b887662f8.png">

israelidanny commented on metaplex/js/test/store.test.ts:
```diff
@@ -0,0 +1,36 @@
+
+  describe('Store', () => {
+    test('SetStore', async () => {
+      const wallet = new NodeWallet(FEE_PAYER);
```
In a proper test this should probably not be the FEE_PAYER but rather a randomly generated wallet? Also not sure I should've copied NodeWallet as is, just FYI.

israelidanny commented on metaplex/js/test/utils.ts:
```diff
@@ -0,0 +1,124 @@
+  ]),
+);
+
+export const STORE_OWNER_PUBKEY = new PublicKey('7hKMAoCYJuBnBLmVTjswu7m6jcwyE8MYAP5hPijUT6nd');
```
Lots of this stuff might not be needed?

israelidanny commented on metaplex/js/package.json:
```diff
@@ -45,7 +52,15 @@
+    "esbuild-runner": "^2.2.1",
     "eslint": "^8.3.0",
-    "rimraf": "^3.0.2"
+    "jest": "^27.4.7",
```
I don't fully understand this sentence 😅 Could you rephrase?

aheckmann replied:
`tape` and `spok` are test dependencies we are using instead of `jest` https://github.com/metaplex-foundation/metaplex-program-library/blob/master/token-metadata/js/package.json#L51

israelidanny replied:
Done

aheckmann commented on metaplex/js/test/setup/build.ts:
```diff
@@ -0,0 +1,73 @@
+    }
+  }
+
+  if (!isLocal) {
```
`isLocal` was so the JS SDK repo could either download MPL code from github or use a local copy. We don't need `isLocal` here since we already have all of the MPL code. Make the `isLocal` logic the default behavior (build the local code) and remove the non-isLocal logic.

israelidanny replied:
Done

aheckmann commented on metaplex/js/test/utils.ts:
```diff
@@ -2,8 +2,9 @@ import path from 'path';
+import { LOCALHOST } from '@metaplex-foundation/amman';
 
-export const NETWORK = 'http://localhost:8899/';
+export const NETWORK = LOCALHOST;
```
🥇

israelidanny replied:
Done

israelidanny commented on metaplex/js/test/utils.ts:
```diff
@@ -0,0 +1,124 @@
+import { tmpdir } from 'os';
+import { readFileSync } from 'fs';
+
+export const NETWORK = 'http://localhost:8899/';
```
Done

israelidanny commented:
@aheckmann Cleaned up all your comments
@austbot Switched to `tape` 
Removed some unnecessary stuff like `NodeWallet` and rewrote the test to be more like what MPL does (as opposed to JS-SDK)

aheckmann commented on metaplex/js/test/setup/build.ts:
```diff
@@ -0,0 +1,27 @@
+    'metaplex/program',
+  ];
+  rmSync(tmpTestDir, { recursive: true, force: true });
+  mkdirSync(tmpTestDir);
```
still needed?

israelidanny replied:
Nice catch :)

aheckmann reviewed (dismissed):
just a question about the tmpTestPath - else LGTM

israelidanny commented:
> just a question about the tmpTestPath - else LGTM

Yup, more junk that I forgot. Removed it now.

aheckmann commented:
will land after CI passes

aheckmann commented:
CI failure. @israelidanny please take a look https://github.com/metaplex-foundation/metaplex-program-library/runs/4957430693?check_suite_focus=true

israelidanny commented:
> CI failure. @israelidanny please take a look https://github.com/metaplex-foundation/metaplex-program-library/runs/4957430693?check_suite_focus=true

![image](https://user-images.githubusercontent.com/1970424/151244868-a6fa1ae2-fab5-4c31-88e3-066a2594f7b9.png)

Looks like it's complaining about package.json changes? Seems to be by design in this PR though?

aheckmann commented:
CI green
